### PR TITLE
Add watcher for ManagedClusterView changes

### PR DESCRIPTION
**Pull request by BenamarMk**
During failover/failback, the VRG state will change from Primary to secondary and vise-versa.
The ManagedClusterView (MCV) is a way to get the resource info from the managed clusters.
The DRPC uses it to get the VRG object in order to inspect its replication state. The reconciler
will watch for the MCV changes, and it will reconcile when it does.

### CODE REVIEW

**Review by ShyamsundarR — APPROVED** [review]
Looks good, no issues here.

The requeue aspects are a bit confusing, from before and now, relooking at the same, but should not affect this patch.

File: controllers/drplacementcontrol_controller.go [hunk 1]
```diff
@@ -197,13 +247,22 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
...
+	logger.Info("Finished processing", "Requeue?", requeue)
+
 	if !requeue {
 		r.Callback(d.instance.Name)
+
+		return ctrl.Result{}, nil
 	}
 
-	logger.Info("Finished processing", "Requeue?", requeue)
+	if d.mcvRequestInProgress {
```
[the remark this answers is not in the document]
Comment (BenamarMk) [reply controllers/drplacementcontrol_controller.go:258]: You are absolutely right.  But, somehow I don't have a good feeling about it, that's why I am requeueing after few second or few minutes depending on which action it is.  I have seen many strange things happening with MCV requests.  At some point we will remove the requeueDuration and just go with notification.

**Comment by ShyamsundarR** [comment]
@BenamarMk there are conflicts, PTAL and then we can merge this if resolved.

**Comment by BenamarMk** [comment]
@ShyamsundarR the conflict has been resolved.  Ready to be merged.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**